### DISCUSSION TITLE
Retrieving commands from app

### DISCUSSION THREAD

**Discussion by ndaskalov-mms** [beeware/toga#3043] [Debugging help]
Hi
I would like to retrieve commands from app and serialize them. The goal is to be able to modify the menu and after to restore it as it was. I tried using .commands property, but it returns also the Separator. What is the best way to achieve this? 
Thanks in advance 
Nik

**Comment by freakboy3742** [comment]
I'm not sure I understand why you'd need to serialise direct from the `commands` property. 

There are commands that are installed by default; you won't need to install those manually on a restore. The only commands you would need to restore are ones you've added yourself... so you've got access to the raw definition at the time you originally added them. In addition to adding the command, add the command to whatever serialisation format you're going to use to persist the menu definitions, and on restore, process all those serialised menu definitions.
